src/NodaTime/Calendars/WeekYearRules.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright 2016 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using JetBrains.Annotations;
+using NodaTime.Extensions;
+using NodaTime.Utility;
+using System;
+using System.Globalization;
+using static System.Globalization.CalendarWeekRule;
+
+namespace NodaTime.Calendars
+{
+    /// <summary>
+    /// Factory methods to construct week-year rules supported by Noda Time.
+    /// </summary>
+    public static class WeekYearRules
+    {
+        /// <summary>
+        /// Returns an <see cref="IWeekYearRule"/> consistent with ISO-8601.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// In the standard ISO-8601 week algorithm, the first week of the year
+        /// is that in which at least 4 days are in the year. As a result of this
+        /// definition, day 1 of the first week may be in the previous year. In ISO-8601,
+        /// weeks always begin on a Monday, so this rule is equivalent to the first Thursday
+        /// being in the first Monday-to-Sunday week of the year.
+        /// </para>
+        /// <para>
+        /// For example, January 1st 2011 was a Saturday, so only two days of that week
+        /// (Saturday and Sunday) were in 2011. Therefore January 1st is part of
+        /// week 52 of week-year 2010. Conversely, December 31st 2012 is a Monday,
+        /// so is part of week 1 of week-year 2013.
+        /// </para>
+        /// </remarks>
+        /// <value>A <see cref="IWeekYearRule"/> consistent with ISO-8601.</value>
+        [NotNull] public static IWeekYearRule Iso { get; } = new SimpleWeekYearRule(4, IsoDayOfWeek.Monday, false);
```

Comment:
These are only `[NotNull]` in master/2.1. In 2.0.x, they're not (see 24fdeef).

Does that matter? (I suspect not, since 2.0.x would be a breaking change from 1.4.x anyway.)